File: backend/app/schemas/session.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel

SessionType = Literal["practice", "lesson", "review"]
SessionSubject = Literal["math", "english", "chinese"]


class SessionCreate(BaseModel):
    session_type: SessionType = "practice"
    subject: SessionSubject = "math"
    knowledge_point_id: UUID | None = None


class SessionResponse(BaseModel):
    id: UUID
    student_id: UUID
    session_type: str
    subject: str
    knowledge_point_id: UUID | None
    started_at: datetime
    ended_at: datetime | None
    duration_sec: int | None
    problems_total: int | None
    problems_correct: int | None
    score_pct: float | None
    xp_earned: int
    summary: str | None

    model_config = {"from_attributes": True}

File: frontend/src/types/session.ts
// Types derived from backend Pydantic schemas:
//   backend/app/schemas/session.py
//   backend/app/models/learning.py

/** Backend: SessionCreate – POST /sessions */
export interface SessionCreate {
  session_type?: string;
  subject?: string;
  knowledge_point_id?: string | null;
}

/** Backend: SessionResponse – GET /sessions/:id, POST /sessions response */
export interface SessionResponse {
  id: string;
  student_id: string;
  session_type: string;
  subject: string;
  knowledge_point_id: string | null;
  started_at: string;
  ended_at: string | null;
  duration_sec: number | null;
  problems_total: number | null;
  problems_correct: number | null;
  score_pct: number | null;
  xp_earned: number;
  summary: string | null;
}

/** Minimal session shape returned by POST /sessions (for page navigation) */
export interface SessionBrief {
  id: string;
  session_type: string;
  subject: string;
  started_at: string;
}
